components/certificate-mgt/org.wso2.carbon.identity.certificate.management.core/src/main/java/org/wso2/carbon/identity/certificate/management/core/CertificateManagementServiceImpl.java

```diff
@@ -0,0 +1,177 @@
+import org.wso2.carbon.identity.core.util.IdentityTenantUtil;
+
+import java.util.UUID;
+
+/**
+ * This class is responsible for managing certificates of a tenant.
+ */
+public class CertificateManagementServiceImpl implements CertificateManagementService {
+
+    private static final Log LOG = LogFactory.getLog(CertificateManagementServiceImpl.class);
+    private static final CertificateManagementService INSTANCE = new CertificateManagementServiceImpl();
+    private static final CertificateManagementDAOImpl DAO = new CertificateManagementDAOImpl();
+    private static final CertificateValidator CERTIFICATE_VALIDATOR = new CertificateValidator();
+
+    private CertificateManagementServiceImpl() {
+    }
+
+    public static CertificateManagementService getInstance() {
+
+        return INSTANCE;
+    }
+
+    /**
+     * Add a certificate.
+     *
+     * @param certificate  Certificate information.
+     * @param tenantDomain Tenant domain.
+     * @return Certificate ID.
+     * @throws CertificateMgtException If an error occurs while adding the certificate.
+     */
+    @Override
+    public String addCertificate(Certificate certificate, String tenantDomain) throws CertificateMgtException {
+
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("Adding certificate with name: " + certificate.getName());
+        }
+        doPreAddCertificateValidations(certificate);
+        String generatedCertificateId = UUID.randomUUID().toString();
+        DAO.addCertificate(generatedCertificateId, certificate, IdentityTenantUtil.getTenantId(tenantDomain));
+        return generatedCertificateId;
+    }
+
+    /**
+     * Get certificate information with given id.
+     *
+     * @param certificateId Certificate ID.
+     * @param tenantDomain  Tenant domain.
+     * @return Certificate information.
+     * @throws CertificateMgtException If an error occurs while getting the certificate.
+     */
+    @Override
+    public Certificate getCertificate(String certificateId, String tenantDomain) throws CertificateMgtException {
+
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("Retrieving certificate with id: " + certificateId);
+        }
+        Certificate certificate = DAO.getCertificate(certificateId, IdentityTenantUtil.getTenantId(tenantDomain));
+        if (certificate == null) {
+            if (LOG.isDebugEnabled()) {
+                LOG.debug("No certificate found for the id: " + certificateId);
+            }
+            throw CertificateMgtUtil.raiseClientException(CertificateMgtErrors.ERROR_CERTIFICATE_DOES_NOT_EXIST,
+                    certificateId);
+        }
+
+        return certificate;
+    }
+
+    /**
+     * Update a certificate with given id.
+     *
+     * @param certificateId Certificate ID.
+     * @param certificate   Certificate content.
+     * @param tenantDomain  Tenant domain.
+     * @throws CertificateMgtException If an error occurs while updating the certificate.
```

Comment:
better mention that null values are ignored.

Reply:
Fixed with https://github.com/wso2/carbon-identity-framework/pull/6043/commits/d53d972086232290a3c952455a77300bdb73d336